server/routes/walletRouter.js

```diff
@@ -22,7 +23,13 @@ router.get(
   '/:wallet_id/trust_relationships',
   handlerWrapper(walletGetTrustRelationships),
 );
-router.post('/', handlerWrapper(walletPost));
+router.post('/', handlerWrapper(async (req, res) => {
+  Joi.assert(req.body, Joi.object({
```

Comment:
This should be in the walletPost handler function, not the router one. Also, extend the validation to ensure only alphanumeric characters.

Reply:
do we also need to contain the character '-'? In the wallet handler test, the test use 'test-wallet-2' as the name. so my new upload below does not pass the test.

Reply:
You can make exceptions for the following characters , `- @ .`